source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -413,6 +419,48 @@ DnsCacheImpl::PrimaryHostInfo::~PrimaryHostInfo() {
   parent_.stats_.num_hosts_.dec();
 }
 
+void DnsCacheImpl::addCacheEntry(const std::string& host,
+                                 const Network::Address::InstanceConstSharedPtr& address) {
+  if (!key_value_store_) {
+    return;
+  }
+  const std::string value = absl::StrCat(address->asString());
+  key_value_store_->addOrUpdate(host, value);
```

Comment:
oh sorry, I could pass in the string view, but I plan on adding TTL.  So this will be used for temporary storage in the immediate follow up and I'd prefer to leave as is?